500 page

VerevkinAlexander opened the pull request:
<h2>General</h2>
Closes #9

Implemented a new page that can be displayed on critical errors.

web-systems-solutions committed:
added: button in footer, modal report, setter and getter for local.storage errors, multylang for button, quary's catchs, componentDidCatch for all pages

yuriikarpenkoo committed:
test commit

yuriikarpenkoo committed:
first commit

yuriikarpenkoo committed:
Merge branch 'staging' into 1696_DebugReport

yuriikarpenkoo committed:
Merge branch 'develop' into 1696_DebugReport

yuriikarpenkoo committed:
add new button for copy errors, take and download screenshort

yuriikarpenkoo committed:
try to fix git errors

yuriikarpenkoo committed:
remove alert, add catch mathod for appInit, add auto read errors on open action of modal report

yuriikarpenkoo committed:
remove all componentDidCatch

yuriikarpenkoo committed:
remove error in footer

yuriikarpenkoo committed:
delete excess code, rename errors to logs actions, add method for catch all console actions

yuriikarpenkoo committed:
add logs for warn, error, info

yuriikarpenkoo committed:
add type to logs

yuriikarpenkoo committed:
add type for query error

yuriikarpenkoo committed:
Merge pull request #6 from blockchainprojects/1696_DebugReport_hooks

1696 debug report hooks

yuriikarpenkoo committed:
add window onerror

yuriikarpenkoo committed:
Merge pull request #7 from blockchainprojects/1696_DebugReport_hooks

add window onerror

yuriikarpenkoo committed:
add page 505, show on window.onerror, add show logs on button in page

yuriikarpenkoo committed:
fix by comments

yuriikarpenkoo committed:
add copy to buffer

yuriikarpenkoo committed:
Merge branch '1696_DebugReport' of https://github.com/blockchainprojects/bitshares-ui into 500_page

yuriikarpenkoo committed:
fix log's view of textarea

yuriikarpenkoo committed:
1696 Removed impacted style

yuriikarpenkoo committed:
check reject methods

yuriikarpenkoo committed:
Merge branch '1696_DebugReport' of https://github.com/blockchainprojects/bitshares-ui into 500_page

Commit:
fixed issue with app rerendering on page 500 appearing

VerevkinAlexander committed:
I used Page500 component as a wrapper for ap ui

Commit:
Merge branch 'develop' into 500_page

Commit:
remove unnecessary changes

Signed-off-by: Stefan Schiessl <stefan.schiessl@blockchainprojectsbv.com>

Commit:
remove unnecessary translations

Signed-off-by: Stefan Schiessl <stefan.schiessl@blockchainprojectsbv.com>

Commit:
one more

Signed-off-by: Stefan Schiessl <stefan.schiessl@blockchainprojectsbv.com>

sschiessl-bcp commented:
I have resolved the conflicts and removed unnecessary code, please observe my last commits on this branch.

I expect that you check every commit such that you do not check in files that have only been reformatted anmd remove debug statements.

VerevkinAlexander committed:
fix Page_500

VerevkinAlexander commented:
@sschiessl-bcp 
I have checked commits after the merge and how "Page_500" component is working after an error is thrown. And I have found out that it is broken. For fixing it: I modified "componentDidCatch" method in "AppInit" component, and rewrote "Page_500" component.

sschiessl-bcp commented on app/App.jsx:
```diff
@@ -265,11 +268,25 @@ class App extends React.Component {
         );
         updateGatewayBackers();
     }
+    static getDerivedStateFromProps(nextProps, prevState) {
```
Why is this necessary?

VerevkinAlexander replied:
Now, its not. I've removed it.

sschiessl-bcp commented on app/App.jsx:
```diff
@@ -143,13 +144,15 @@ class App extends React.Component {
             incognitoWarningDismissed: false,
-            height: window && window.innerHeight
+            height: window && window.innerHeight,
+            errorModule: false
```
Is this used within App.jsx?

VerevkinAlexander replied:
Now, its not. I've removed it.

sschiessl-bcp commented on app/App.jsx:
```diff
@@ -335,9 +352,20 @@ class App extends React.Component {
+            theme,
+            location,
+            match,
+            isErrorCaught,
```
Is this even set within App.jsx?

VerevkinAlexander replied:
It's set in AppInit's componentDidCatch.

VerevkinAlexander committed:
add clearCaughtError, remove unnecessary code
